js/classic/ui.js

```diff
@@ -9,6 +9,223 @@
+    this._hideableChildren = [];
+    this._isExpanded = false;
+    this._selector = $(window.document.createElement('div'));
+    this._selector.addClass('classic-ui-block');
+
+    this.isExpanded = (isExpanded !== undefined ? isExpanded : true);
+  };
+
+  ui.Block.prototype = {
+    /**
+     * Returns true if this block is expanded.
+     */
+    get isExpanded() { return this._isExpanded; },
+
+    /**
+     * Sets whether this block is expanded or not.
+     *
+     * @param state true for expanding this block.
+     */
+    set isExpanded(state) {
+      this._isExpanded = state;
+      for (var i = 0; i < this._hideableChildren.length; ++i) {
+        this._hideableChildren[i].css('display', state ? '' : 'none');
+      }
+    },
+
+    /**
+     * Toggles the state of whether this block is expanded or not.
+     */
+    toggleExpand: function() { this.isExpanded = !this.isExpanded; },
+
+    /**
+     * Returns the jquery selector representing this block ui component.
+     */
+    get selector() { return this._selector; },
+
+    /**
+     * Adds an jquery selector as a child of this ui component.
+     *
+     * @param selector the jquery selector to be added.
+     * @param index the position to insert.
+     * @param hideable whether this child selector should hide when this block
+     *     is not expanded or not.
+     */
+    addChildSelector: function(selector, index, hideable) {
+      index = Math.max(0, index);
+      if (this._selector.children().length <= index) {
+        this._selector.append(selector);
+      } else {
+        this._selector.children()[index].before(selector);
+      }
+      if (hideable) {
+        this._hideableChildren.push(selector);
+        if (!this._isExpanded) {
+          selector.css('display', 'none');
+        }
+      }
+    },
+
+    /**
+     * Destroys the whole ui components, which includes all the children.
+     */
+    destroy: function() { this._selector.remove(); }
+  };
+
+  ui.SwitchButtonBlock = function(
+      handler, name, description, defaultState, enable) {
+    ui.Block.call(this, false);
+    var title = $(window.document.createElement('div'));
+    var titleLeft = $(window.document.createElement('div'));
+    var titleRight = $(window.document.createElement('div'));
+    title.addClass('switch-button-title');
+    titleLeft.addClass('left');
+    titleRight.addClass('right');
+    title.append(titleLeft).append(titleRight);
+    this._button = new ui.SwitchButton(handler, defaultState, enable);
+    titleLeft.append(this._button.selector);
+    titleRight.html(name);
+    titleRight.click(this.toggleExpand.bind(this));
+
+    var body = $(window.document.createElement('div'));
+    body.addClass('switch-button-body').html(description);
+
+    this.addChildSelector(title, 0, false);
+    this.addChildSelector(body, 1, true);
+  };
+
+  ui.SwitchButtonBlock.State = ui.SwitchButton.State;
+
+  ui.SwitchButtonBlock.prototype = {
+    State: ui.SwitchButtonBlock.State,
```

Comment:
It seems like a reference to constant value. I'll suggest you rename it to `STATE` because we have property name `state` which is confusing.